students/ian_letourneau/Lesson03/slicing_lab.py

```diff
@@ -0,0 +1,99 @@
+## Ian Letourneau
+## 4/26/2018
+## A script with various sequencing functions
+
+def exchange_first_last(seq):
+    """A function to exchange the first and last entries in a sequence"""
+    middle = list(seq)
+    middle[0] = seq[-1]
+    middle[-1] = seq[0]
+
+    if type(seq) is str:
+        return "".join(middle)
+    elif type(seq) is tuple:
+        return tuple(middle)
+    else:
+        return middle
+
+def remove_every_other(seq):
+    """A function to remove every other entry in a sequence"""
+    middle = []
+    for index in range(len(seq)):
+        if not index%2 or index == 0:
+            middle.append(seq[index])
+
+    if type(seq) is str:
+        return "".join(middle)
+    elif type(seq) is tuple:
+        return tuple(middle)
+    else:
+        return middle
+
+def remove_four(seq):
+    """A function that removes the first 4 and last four entries in a sequence,
+        and then removes every other entry in the remaining sequence"""
+    middle = list(seq)
+    del middle[0:4]
+    del middle[-4:len(seq)]
+
+    if type(seq) is str:
+        return "".join(remove_every_other(middle))
+    elif type(seq) is tuple:
+        return tuple(remove_every_other(middle))
+    else:
+        return remove_every_other(middle)
+
+def reverse(seq):
+    """A function that reverses a seq"""
+    middle = list(seq)
+    count = -1
+    for index in range(len(seq)):
+        middle[index] = seq[count]
+        count -= 1
+
+    if type(seq) is str:
+        return "".join(middle)
+    elif type(seq) is tuple:
+        return tuple(middle)
+    else:
+        return middle
+
+def thirds(seq):
+    """A function that splits a sequence into thirds,
+            then returns a new sequence using the last, first, and middle thirds"""
+    middle = list(seq)
+    one = middle[0:int(len(seq)/3)]
+    two = middle[int(len(seq)/3):int(len(seq)/3*2)]
+    three = middle[int(len(seq)/3*2):len(seq)]
+    a_new_sequence = three + one + two
+
+    if type(seq) is str:
+        return "".join(a_new_sequence)
+    elif type(seq) is tuple:
+        return tuple(a_new_sequence)
+    else:
+        return a_new_sequence
```

Comment:
same comment for all functions, should be using slicing only (`thirds` does use slicing) and no type checking needed.